ui/app/components/console/ui-panel.js

```diff
@@ -61,23 +79,43 @@ export default Ember.Component.extend({
       this.logAndOutput(command, inputError);
       return;
     }
-    let serviceFn = service[method];
-    serviceFn
-      .call(service, path, data, flags.wrapTTL)
-      .then(resp => {
-        this.logAndOutput(command, logFromResponse(resp, path, method, flags));
-      })
-      .catch(error => {
-        this.logAndOutput(command, logFromError(error, path, method));
-      });
+    this.get('runner').perform(
+      () => service[method].call(service, path, data, flags.wrapTTL),
+      resp => this.logAndOutput(command, logFromResponse(resp, path, method, flags)),
+      error => this.logAndOutput(command, logFromError(error, path, method))
+    );
   },
 
+  refreshRoute() {
+    let owner = getOwner(this);
+    let routeName = this.get('router.currentRouteName');
+    let route = owner.lookup(`route:${routeName}`);
+
+    this.get('runner').perform(
+      () => route.refresh(),
+      () => this.logAndOutput(null, { type: 'success', content: 'The current screen has been refreshed!' }),
+      () =>
+        this.logAndOutput(null, {
+          type: 'error',
+          content: 'The was a problem refreshing the current screen.',
+        })
+    );
```

Comment:
I think it's unusual to mix promise callbacks with ember-concurrency. If the `runner` task took one argument and `yield`s it, this could be written as:

```js
try {
  yield this.get('runner').perform(() => route.refresh());
  this.logAndOutput(null, { type: 'success', content: 'The current screen has been refreshed!' });
} catch(err) {
  this.logAndOutput(null, { type: 'error', content: 'The was a problem refreshing the current screen.' })
}
```

Similarly, if `executeCommand` was an e-c task, you could drop the wrapper action and using try-catch there, too.

The try-catch approach to error handling is one of my favourite things about e-c tasks, as I think the more linear code flow is easier to understand. But it's all personal preference!

Reply:
Thank you so much for this - I had a feeling the runner was weird but couldn't put my finger on it. Making `refreshRoute` a task and then removing runner cleans things up quite nicely!